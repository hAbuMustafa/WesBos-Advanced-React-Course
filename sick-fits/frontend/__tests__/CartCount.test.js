import { render, waitFor } from '@testing-library/react';
import wait from 'waait';
import CartCount from '../components/CartCount';

describe('<CartCount/>', () => {
  it('Renders', () => {
    render(<CartCount count={10} />);
  });

  it('Matches Snapshot', () => {
    const { container } = render(<CartCount count={17} />);
    expect(container).toMatchSnapshot();
  });

  it('Update via props', async () => {
    const { container, rerender, debug } = render(<CartCount count={17} />);
    expect(container.textContent).toBe('17');
    rerender(<CartCount count={25} />);
    await wait(2000);
    expect(container.textContent).toBe('25');
    expect(container).toMatchSnapshot();
  });
});
