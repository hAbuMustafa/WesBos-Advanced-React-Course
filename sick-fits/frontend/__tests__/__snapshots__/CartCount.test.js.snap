// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount/> Matches Snapshot 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-xxvp4g-1 gwXiGr"
  >
    <div>
      <div
        class="CartCount__Dot-sc-xxvp4g-0 hfmEXi count"
      >
        17
      </div>
    </div>
  </span>
</div>
`;

exports[`<CartCount/> Update via props 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-xxvp4g-1 gwXiGr"
  >
    <div>
      <div
        class="CartCount__Dot-sc-xxvp4g-0 hfmEXi count count-enter-done"
      >
        25
      </div>
    </div>
  </span>
</div>
`;
